amazon-kinesis-client/src/main/java/software/amazon/kinesis/common/HashKeyRangeForLease.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.kinesis.common;
+
+import com.google.common.base.Joiner;
+import lombok.Data;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang3.Validate;
+import software.amazon.awssdk.services.kinesis.model.HashKeyRange;
+
+@Data
+@Accessors(fluent = true)
+/**
+ * Lease POJO to hold the starting hashkey range and ending hashkey range of kinesis shards.
+ */
+public class HashKeyRangeForLease {
```

Comment:
Is there a reason we need to maintain the range? Can we just have two separate DDB fields, StartingHashKey and EndingHashKey so we can avoid doing this parsing altogether?

Reply:
Just wanted to make both the updates atomic.

Reply:
The updates to DDB are atomic whether you do on single field or two fields in DDB. I would prefer having it as two fields in DDB and avoid this parsing logic.  (Having it as range stops us to create indexes if needed in future)

Reply:
Not the DDB updates. I wanted this to be updated and maintained in a pair across the KCL code.

Reply:
I will check if we can have different fields. But I dont see if we might introduce indices to KCL lease table.